packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterInputMultipleSingleSelect.tsx

```diff
@@ -0,0 +1,152 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import Autocomplete, { AutocompleteProps, createFilterOptions } from '@mui/material/Autocomplete';
+import Chip from '@mui/material/Chip';
+
+import TextField from '@mui/material/TextField';
+import { unstable_useId as useId } from '@mui/material/utils';
+import { GridFilterItem } from '../../../models/gridFilterItem';
+
+export type GridFilterInputMultipleSingleSelectProps = {
+  item: GridFilterItem;
+  applyValue: (value: GridFilterItem) => void;
+  // Is any because if typed as GridApiRef a dep cycle occurs. Same happens if ApiContext is used.
+  apiRef: any;
+  focusElementRef?: React.Ref<any>;
+  type?: 'singleSelect';
+} & Omit<AutocompleteProps<any[], true, false, true>, 'options' | 'renderInput'>;
+
+const getSingleSelectOptionFormatter =
+  ({ valueFormatter, field }, api) =>
+  (option) => {
+    if (typeof option === 'object') {
+      return option.label;
+    }
+    return valueFormatter && option !== '' ? valueFormatter({ value: option, field, api }) : option;
+  };
+
+const getValueFromOption = (option) => (typeof option === 'object' ? option.value : option);
+
+const isOptionEqualToValue = (option, value) =>
+  getValueFromOption(option) === getValueFromOption(value);
+
+const filter = createFilterOptions<any>();
+
+function GridFilterInputMultipleSingleSelect(props: GridFilterInputMultipleSingleSelectProps) {
+  const { item, applyValue, type, apiRef, focusElementRef, ...other } = props;
+  const id = useId();
+
+  const currentColumn = item.columnField ? apiRef.current.getColumn(item.columnField) : null;
+  const currentValueOptions = React.useMemo(() => {
```

Comment:
Yes, it will be more logic, since data are resolved from the apiRef. I've done the modification for the following namings:
- resolvedColumn
- resolvedValueOptions
- resolvedFormattedValueOptions